### PR TITLE
Rework snap hooks

### PR DESCRIPTION
**Pull request by jardon**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
The snap configuration is triggered on every config changed event.

* **What is the new behavior (if this is a feature change)?**
The initial snap configuration is only called on install which frees us up to implement configuration event specific events.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

* **Other information**:
N/A

### CODE REVIEW

**Comment by jardon** [comment]
Pushed these changes to `8.0/edge/fix-configure-test` for testing.

File: snap/hooks/install [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+echo "log_bin_index = $CONFIG_DIR/binlog.index" >> $CONFIG_DIR/mysql.conf.d/mysqld.cnf
+echo "mysqlx_socket = $CONFIG_DIR/mysqlx.sock" >> $CONFIG_DIR/mysql.conf.d/mysqld.cnf
+
+$SNAP/usr/sbin/mysqld --initialize --datadir=$CONFIG_DIR/data --user=root
+
+# Add default client config for exporter
+sed -i '32s/^/\n[client]\nuser = exporter\npassword = exporterpasswd\nport = 3306\nhost = localhost\n\n/' $CONFIG_DIR/mysql.cnf
+
+# Change ownership of snap directories to allow snap_daemon to read/write
+chown -R 584788:root $SNAP_DATA/*
```
Comment (paulomach) [inline snap/hooks/install:30]: who is `584788` ?

**Review by paulomach — APPROVED** [review]
Left a question, but lgtm

**Review by shayancanonical — APPROVED** [review]

File: snap/hooks/install (hunk 1 above)
Comment (jardon) [reply snap/hooks/install:30]: That is the `snap_daemon` user id.  This change is to help with some changes to the associated rock.

**Pull request merged**